music_assistant/server/providers/apple_music/__init__.py

```diff
@@ -199,9 +199,16 @@ async def get_library_tracks(self) -> AsyncGenerator[Track, None]:
         """Retrieve library tracks from the provider."""
         endpoint = "me/library/songs"
         song_catalog_ids = []
-        for item in await self._get_all_items(endpoint, include="catalog"):
-            if item and "catalog" in item["relationships"]:
-                song_catalog_ids.append(item["relationships"]["catalog"]["data"][0]["id"])
+        for item in await self._get_all_items(endpoint):
+            catalog_id = item.get("attributes", {}).get("playParams", {}).get("catalogId")
+            if not catalog_id:
+                self.logger.warning(
```

Comment:
Does it need to be warning level ? Maybe just use debug level ?

Reply:
I was thinking in the lines of how we also warn about songs without artist. The result the catalog_id not being available is that the song does not show up in MA, so I thought a warning was appropriate. If you think it should be debug, I will change it.

Reply:
From what I read between the lines of the issue reports is that this happens for some songs that folks uploaded to their library ages ago so maybe debug is just enough. But do that in a follow up PR if folks complain about noisy logs.